modules/nf-core/mageck/count/main.nf

```diff
@@ -0,0 +1,47 @@
+
+process MAGECK_COUNT {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::mageck=0.5.9" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mageck:0.5.9--py37h6bb024c_0':
+        'quay.io/biocontainers/mageck:0.5.9--py37h6bb024c_0' }"
+
+    input:
+    tuple val(meta), path(inputfile)
+    path(library)
+
+    output:
+    tuple val(meta), path("*count*.txt"), emit: count
+    tuple val(meta), path("*.count_normalized.txt"), emit: norm
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def input_file = ("$inputfile".endsWith(".fastq.gz")) ? "--fastq ${inputfile}" :
+        ("$inputfile".endsWith(".txt")) ? "-k ${inputfile}" : ("$inputfile".endsWith(".csv")) ? "-k ${inputfile}" :
+        ("$inputfile".endsWith(".tsv")) ? "-k ${inputfile}"  : ''
```

Comment:
```suggestion
    def input_file = ("$inputfile".endsWith(".fastq.gz")) ? "--fastq ${inputfile}" : "-k ${inputfile}"
```

or not? I suppose withut any input this module wouldn't work anyways?

Reply:
in the meta yml you also allow for fq.gz,fq,and fastq , I guess those should also run with `--fastq`. Could this be captured with a regex in the èndsWith`statement?